lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3196,7 +3198,7 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       //   adj[x] = struct (0, ..., key': adj[y], ..., 0)
       // where `key'` is the field in the cotangent space corresponding to
       // `key`.
-      auto structTy = sei->getOperand()->getType().getASTType();
+      auto structTy = remapType(sei->getOperand()->getType()).getASTType();
```

Comment:
Move line `auto av = getAdjointValue(sei);` above this and use `av.getASTType()` instead of remapping.

Reply:
Actually, nvm. I think you still need a remap here.

Reply:
Actually, nvm. I think you still need a remap here.